### ISSUE TITLE
Can you give tips about finetuning on custom datasets?

### ISSUE THREAD

**Issue by davidefiocco** [grammarly/gector#61]
I have created `train.csv` and `dev.tsv` sets with the aim of finetuning GECToR to do better corrections on a (custom) test set.
Still, I would like to build on the training on which GECToR is based on, and not train from scratch (I have ~50k sentences to finetune on)

For the purpose, I run GECToR with 

    python gector/train.py --transformer_model bert 
                                         --train_set train.csv 
                                         --dev_set dev.csv 
                                         --model_dir . 
                                         --vocab_path ./gector/data/output_vocabulary
                                         --special_tokens_fix 0

starting from the pre-trained encoder https://grammarly-nlp-data-public.s3.amazonaws.com/gector/bert_0_gector.th in the working directory.

Is this going in the good direction?
Is the reuse of a your `output_vocabulary` needed to avoid catastrophical forgetting? (I tried to understand this from https://github.com/grammarly/gector/issues/40#issuecomment-671023748 but it's not clear to me yet).

Are there parameters in `training.py` I should really pay attention to and use? Thanks!

**Comment by skurzhanskyi** [comment]
Looks good for me.
Using `output_vocabulary` is necessary as you use pre-training. Pre-training doesn't make sense without the vocabulary.
I would also suggest to take a look at parameters related to training schedule like `lr`, `updates_per_epoch`.

**Comment by mina1460** [comment]
hi @skurzhanskyi,
I don't understand how is this command going to make use of any of the pre-trained models if he didn't use the --pretrain and --pretrain _folder. Can you elaborate on this please? or was this an answer based on an older version of the code?
